aws/resource_aws_route53_health_check.go

```diff
@@ -128,6 +128,12 @@ func resourceAwsRoute53HealthCheck() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
+				// The max length of the reference name is 64 characters for the API.
+				// Terraform appends a 26-character unique ID to the provided
+				// reference_name. This limits the length of the resource argument to 38.
+				//
+				// Example generated suffix: -terraform-20190122200019880700000001
+				ValidateFunc: validation.StringLenBetween(0, (64 - resource.UniqueIDSuffixLength)),
```

Comment:
The validation should be subtracting 11 still 😉 `resource.UniqueIDSuffixLength` only represents the digits after the `-terraform-` part that's being added.

```suggestion
				// The max length of the reference name is 64 characters for the API.
				// Terraform appends a 37-character unique ID to the provided
				// reference_name. This limits the length of the resource argument to 27.
				//
				// Example generated suffix: -terraform-20190122200019880700000001
				ValidateFunc: validation.StringLenBetween(0, (64 - resource.UniqueIDSuffixLength - 11)),
```